test/app.listen.js

```diff
@@ -15,4 +16,18 @@ describe('app.listen()', function(){
       done();
     });
   })
+
+  it('should callback on HTTP server errors', function(done){
+    var app = express();
+    var app2 = express();
+
+    var server = app.listen(9999, function(err){
```

Comment:
Can this port number be made dynamic? Getting some race conditions where the Express suite is running multiples times on the same machine and they all want to use port 9999. Just have `app` listen randomly and have `app2` listen to whatever port `app` is listening on should work.